modules/auxiliary/gather/ldap_esc1_cert_finder.rb

```diff
@@ -0,0 +1,420 @@
+    if (value & 0x001F01FF) == 0x001F01FF
+      raw_string += 'FILE_ALL_ACCESS|'
+    else
+      if (value & 0x001200A0) == 0x001200A0
+        raw_string += 'FILE_EXECUTE|'
+      end
+      if (value & 0x00120116) == 0x00120116
+        raw_string += 'FILE_WRITE|'
+      end
+      if (value & 0x00120089) == 0x00120089
+        raw_string += 'FILE_READ|'
+      end
+    end
+    if (value & 0x000F003F) == 0x000F003F
+      raw_string += 'KEY_ALL_ACCESS|'
+    else
+      if (value & 0x00020019) == 0x00020019
+        raw_string += 'KEY_READ|KEY_EXECUTE|'
+      end
+      if (value & 0x00020006) == 0x00020006
+        raw_string += 'KEY_WRITE|'
+      end
+    end
+    if (value & 0x00000100) == 0x00000100
+      raw_string += 'CONTROL_ACCESS|'
+    end
+    if (value & 0x00000080) == 0x00000080
+      raw_string += 'LIST_OBJECT|'
+    end
+    if (value & 0x00000040) == 0x00000040
+      raw_string += 'DELETE_TREE|'
+    end
+    if (value & 0x00000020) == 0x00000020
+      raw_string += 'WRITE_PROPERTY|'
+    end
+    if (value & 0x00000010) == 0x00000010
+      raw_string += 'READ_PROPERTY|'
+    end
+    if (value & 0x00000008) == 0x00000008
+      raw_string += 'SELF_WRITE|'
+    end
+    if (value & 0x00000004) == 0x00000004
+      raw_string += 'LIST_CHILDREN|'
+    end
+    if (value & 0x00000002) == 0x00000002
+      raw_string += 'DELETE_CHILD|'
+    end
+    if (value & 0x00000001) == 0x00000001
+      raw_string += 'CREATE_CHILD|'
+    end
+
+    raw_string.chomp('|')
+  end
+
+  def parse_dacl_or_sacl(acl)
+    flag_allowed_to_enroll = nil
+    allowed_sids = []
+    acl.aces.each do |ace|
+      ace_header = ace[:header]
+      ace_body = ace[:body]
+      ace_string = find_acl_type_string(ace_header[:ace_type])
+      ace_access_mask = parse_access_mask(ace_header[:ace_type], ace_body[:access_mask])
+      print_status("   ACE Type: #{ace_string}")
+      if ace_header[:ace_flags][:inherit_only_ace] == 1
+        print_warning('      ACE only affects those that inherit from it, not those that it is attached to. Ignoring this ACE, as its not relevant.')
+        next
+      elsif ace_header[:ace_flags][:inherited_ace] == 1
+        print_status('      ACE was inherited!')
+      end
+      print_status("      ACE Access Mask (Parsed): #{ace_access_mask}")
+      print_status("      ACE Access Mask (Raw): 0x#{ace_body[:access_mask].to_s.to_i.to_s(16).upcase}")
+      print_status("      SID: #{ace_body[:sid]}")
+      # To decode the ObjectType we need to do another query to CN=Configuration,DC=daforest,DC=com
+      # and look at either schemaIDGUID or rightsGUID fields to see if they match this value.
+      if ace_body[:flags] && ace_body[:flags][:ace_object_type_present] == 1
+        object_type = ace_body[:object_type]
+
+        if ace_access_mask.match(/CONTROL_ACCESS/) && (object_type == '0e10c968-78fb-11d2-90d4-00c04f79dc55' || object_type == 'a05b8cc2-17bc-4802-a710-e7c15ab866a2')
```

Comment:
Constants for the UUIDs were implemented in https://github.com/rapid7/metasploit-framework/pull/17122/commits/d025fe6830cda7282cd3a70d7700773dbb6329ec. As for the access mask issue, I'm awaiting Spencer's PR to land with his updates on that before I address any access mask related issues so that I can rebase my changes on top of his work.